### DISCUSSION TITLE
Multiple JK BMS BLE Connections

### DISCUSSION THREAD

**Discussion by ecornwell01** [syssi/esphome-jk-bms#533] [Q&A] [answered]
Hello,

I'm very new to the idea of Home Assistant and ESPHome.  I've got a solar system setup and I'd like to monitor my JK BMS's with BLE.  My home assistant install isn't near my batteries so I found this project and have been working on getting it working with my JK_PB2A16S20P.  Right now I have 4 of them but eventually I'll have 6.  I found the example yaml file for multiple and it has 3 so I duplicated it for my 4 and got an error when I validated the file.
```
Failed config

ble_client: [source /config/esphome/esp32-bluetooth-proxy-166694.yaml:81]
  
  Component ble_client supports a maximum of 3 entries (4 found).
```

Is that a functional limitation?  If so, is there another way to monitor all 4?  I've already got them in closed loop communication to my inverter and I monitor them with Solar Assistant but I'd like to be able to more closely monitor the individual cells in Home Assistant. 

Thank you!

**Comment by syssi** [comment, marked as answer]
This limitation is an ESP-IDF recommendation because of the limited resources of an ESP32 WROOM. But there are more powerful ESPs where this limit doesn't apply anymore. The limit is enforced by ESPHome and is not part of this project.

**Reply by syssi** [reply]
See https://github.com/esphome/esphome/pull/7088

**Reply by ecornwell01** [reply]
Thanks for the response!  I think I'm going to try your parallel inverter based code instead.  I've ordered the serial TTL to RS485.
